Concentus.Oggfile/OpusOggReadStream.cs

```diff
@@ -128,37 +125,82 @@ private bool Initialize()
         {
             try
             {
-                OggContainerReader reader = new OggContainerReader(_inputStream, true);
-                if (!reader.Init())
+                var oggContainerReader = new OggContainerReader(_stream, true);
+                if (!oggContainerReader.Init())
                 {
-                    _lastError = "Could not initialize stream";
+                    LastError = "Could not initialize stream";
                     return false;
                 }
 
-                //if (!reader.FindNextStream())
-                //{
-                //    _lastError = "Could not find elementary stream";
-                //    return false;
-                //}
-                if (reader.StreamSerials.Length == 0)
+                if (oggContainerReader.StreamSerials.Length == 0)
                 {
-                    _lastError = "Initialization failed: No elementary streams found in input file";
+                    LastError = "Initialization failed: No elementary streams found in input file";
                     return false;
                 }
 
-                int streamSerial = reader.StreamSerials[0];
-                _packetProvider = reader.GetStream(streamSerial);
+                int firstStreamSerial = oggContainerReader.StreamSerials[0];
+                _packetProvider = oggContainerReader.GetStream(firstStreamSerial);
+
+                GranuleCount = _packetProvider.GetGranuleCount();
+                PageCount = _packetProvider.GetTotalPageCount();
+
                 QueueNextPacket();
 
                 return true;
             }
             catch (Exception e)
             {
-                _lastError = "Unknown initialization error: " + e.Message;
+                LastError = "Unknown initialization error: " + e.Message;
                 return false;
             }
         }
 
+        /// <summary>
+        /// Seeks the Opus stream for a valid packet at the specified offset.
+        /// </summary>
+        /// <param name="offset">The sample offset.</param>
+        public void SeekTo(long offset)
```

Comment:
What offset is this? File byte offset?

Reply:
Currently it's `granuleOffset` (I did update the code)

However, if it makes more sense to allow SampleOffset, then I need to change this code and also add a Length property instead of GranuleCount

Reply:
Or rename the method to `SeekToGranulePosition` / `SeekToGranuleOffset` ?

Reply:
Or just be even more on the nose, `SeekTo(TimeSpan playbackTime)` and then convert that to granule position with `TimeSpan.TotalSeconds * 48000`

Reply:
Then I would propose three methods:
1. SeekToGranulePosition(long granulePosition)
2. SeekTo(long samplePosition)
3. SeekTo(TimeSpan playbackTime)

I hope I have enough information present in this class or the decoder to do all the conversion...

Reply:
I don't believe any consumer of the code would know or care about granule position so I see no reason to expose it. Also it's not possible to seek to an exact sample since you can only seek to a single packet which would typically contain 20ms of audio at a time. So seeking to a TimeSpan playback time while making it clear to the user that the seek is only approximate seems like the only reasonable function to implement